[siemensrds] Siemens Smart Thermostat Binding -- Initial contribution (#5739)
openhab/openhab2-addons: bundles/org.openhab.binding.siemensrds/README.md

| Configuration Parameter | Description
|-------------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| userEmail               | The e-mail address of the user account on the cloud server; as entered in the Siemens App when first registering a thermostat.                                      |
| userPassword            | The password of the user account on the cloud server; as entered in the Siemens App.                                                                                |
| pollingInterval         | Time interval in seconds between polling requests to the cloud server; the Default (recommended) interval is set a 60 seconds.                                      |
| apiKey                  | The key code needed to access the application program interface on the Siemens Climatix IC cloud server; you can request a key code from Siemens on their web-site. |

## Thing Configuration for "RDS Smart Thermostat"

> The Climatix IC Account Thing connects to the Siemens Climatix IC cloud server (bridge) to communicate with any respective RDS Smart Thermostat Things associated with that account. Each such RDS Smart Thermostat Thing is identified by means of a unique Plant Id code. The Plant Id code is automatically discovered by the Climatix IC Account Thing.   

Reviewer: ```suggestion
The Climatix IC Account connects to the Siemens Climatix IC cloud server (bridge) to communicate with any respective RDS Smart Thermostats associated with that account.
Each such RDS Smart Thermostat is identified by means of a unique Plant Id code.
The Plant Id code is automatically discovered by the Climatix IC Account.   
```

Author: Ok. Modified.

| Configuration Parameter | Description                                                                                                 | 
|-------------------------|-------------------------------------------------------------------------------------------------------------|
| plantId                 | The unique code to identify a specific RDS Smart Thermostat Thing on the Siemens Climatix IC cloud server.  |

> Note: Do NOT attempt to manually change or enter the Plant Id code; the Climatix IC Account Thing can only connect to the cloud server for RDS Smart Thermostat devices having a valid Plant Id code.         

Reviewer: ```suggestion
```

Author: Ok. Deleted.

## Channels for RDS Smart Thermostat

The RDS Smart Thermostat supports several channels as shown below. 

| Channel                  | Data Type | Description                                                                 |
|--------------------------|-----------|-----------------------------------------------------------------------------|

> | roomTemperature          | Number    | Actual Room Temperature                                                     |

Author: Ok. Done

| thermostatOutputState    | String    | Status of whether the thermostat is Heating, Cooling, or Neither            |
| roomHumidity	           | Number    | Actual Room Humidity                                                        |
| roomAirQuality           | String    | Actual Room Air Quality (Poor..Good)                                        |
| outsideTemperature       | Number    | Actual Outside temperature                                                  |
| greenLeafScore           | String    | Green Leaf Mode / Energy saving level (Poor..Excellent)                     |
| occupancyModePresent     | Switch    | The Thermostat is in the Present Occupancy Mode (Off=Absent, On=Present)    |
| thermostatAutoMode       | Switch    | The Thermostat is in Automatic Mode (Off=Manual, On=Automatic)              |
| hotWaterAutoMode         | Switch    | The Domestic Water Heating is in Automatic Mode (Off=Manual, On=Automatic)  |
| hotWaterSwitchState      | Switch    | The On/Off state of the domestic water heating                              |

## Full Example

### `demo.things` File

> As a general rule it is **recommended** to create the Thing for Climatix IC Account using the PaperUI; however if you wish to do so, you *MAY* create the account thing by means of a `.things` file as follows..

Reviewer: ```suggestion
```

Author: Ok. Deleted

```
Bridge siemensrds:climatixic:mybridgename "Climatix IC Account" [ userEmail="email@example.com", userPassword="secret", apiKey="32-character-code-provided-by-siemens", pollingInterval=60 ]
}
```

> Note: it is **strongly** recommended **NOT** to create RDS Smart Thermostat Things through manual configuration files. The reason is that to configure an RDS Smart Thermostat Thing requires advance knowledge of the "Plant Id" which is a unique code used to identify a specific thermostat device in the Siemens Climatix IC cloud server account. The PaperUI automatic Discovery service (see above) discovers the unique "Plant Id" codes during the discovery process. But if you wanted to create an RDS Smart Thermostat by means of a manual configuration file you would not know its Plant Id. *Nevertheless, for completeness sake, the following shows an example of a manual configuration file, assuming that you had determined the Plant Ids by previously using the auto discovery service*..        

Reviewer: Please have a line break after every sentence.
Don't do recommendations here. As discussed, both ways are equally valid. Just point out that a plantId is required and that this can simplest be obtained through auto-discovery.

Author: Ok. Done

# Siemens RDS Binding

The Siemens RDS binding allows you to connect openHab to Siemens Climtix IC cloud server and integrate your RDS smart thermostats onto the bus.

> ## Binding Configuration

Author: I did use the official template, so I am surprised by your comment.
=> Can you please be more specific about what you don't like?

> ## Binding Configuration

Reviewer: Just a few points:
- What you have in "Binding Configuration" belongs into "Thing Configuration". 
- A section "Item Configuration" does not exist in the template.
- The "Channels" section is missing - this is the most important one (probably what you have as Items right now).
- Every README must have the "Full Example" section.

> ## Binding Configuration

Author: I've done everything except the "Full Example" -- as there is nothing to show an example of. Or do you want (say) a PaperUI screenshot ?

> ## Binding Configuration

Reviewer: Please have a look at other bindings. A "Full Example" shows an example of a thing-file, an item-file and a sitemap-file that fit together.

> ## Binding Configuration

Author: Hmm. I have been testing the binding using OH2 PaperUI, so I have not yet tried to create manual Thing/Item/Sitemap files.

I am reluctant to present example Thing files. I guess the Cloud Account Thing could be created by a file (e.g. putting the login credentials in a file) but I am pretty sure that Smart Thermostat Things can (must) only be created by Automatic Discovery. This is because Auto Discovery finds all Smart Thermostats according to their unique Plant Id codes in the cloud server account, and if the user would try to manually create such a Thing, they have no way to know what Plant Ids are existing in the account, and if they were to just guess a Plant Id that is not in the account, it would fail).

On the other hand, I can more or less understand the need for manual Item an Sitemap files, so I could probably create some examples of that.

> ## Binding Configuration

Reviewer: Every binding must support textual configuration and the examples are crucial for it.
If there is no way for the user to find out the PlantId on his account, he can still use discovery once for that purpose and then go on and configure the Things in a text file.

> ## Binding Configuration

Author: Ok. I have added a Full Example to the ReadMe file. And while working on that I realised that that some of the naming of Channels could be improved for easier understanding, so I also changed the respective Java and XML files to match.
